test/RibbonThetaSTETHVault.ts

```diff
@@ -0,0 +1,2943 @@
+        );
+        assert.equal(
+          auctionDetails.minimumBiddingAmountPerOrder.toString(),
+          "1"
+        );
+        assert.equal(auctionDetails.isAtomicClosureAllowed, false);
+        assert.equal(
+          auctionDetails.feeNumerator.toString(),
+          feeNumerator.toString()
+        );
+        assert.equal(auctionDetails.minFundingThreshold.toString(), "0");
+        assert.equal(
+          await gnosisAuction.auctionAccessManager(currentAuctionCounter),
+          constants.AddressZero
+        );
+        assert.equal(
+          await gnosisAuction.auctionAccessData(currentAuctionCounter),
+          "0x"
+        );
+
+        const initialAuctionOrder = decodeOrder(
+          auctionDetails.initialAuctionOrder
+        );
+
+        const oTokenSellAmount = params.expectedMintAmount
+          .mul(feeDenominator)
+          .div(feeDenominator.add(feeNumerator));
+
+        const oTokenPremium = wmul(
+          (
+            await optionsPremiumPricer.getPremium(
+              await nextOption.strikePrice(),
+              await nextOption.expiryTimestamp(),
+              params.isPut
+            )
+          )
+            .mul(await vault.premiumDiscount())
+            .div(1000),
+          await collateralContract.stEthPerToken()
+        );
+
+        assert.equal(
+          initialAuctionOrder.sellAmount.toString(),
+          oTokenSellAmount.toString()
+        );
+        assert.equal(
+          initialAuctionOrder.buyAmount.toString(),
+          wmul(oTokenSellAmount.mul(BigNumber.from(10).pow(10)), oTokenPremium)
+            .div(BigNumber.from(10).pow(18 - tokenDecimals))
+            .toString()
+        );
+
+        // Hardcoded
+        // assert.equal(auctionDetails.interimSumBidAmount, 0);
+        // assert.equal(auctionDetails.interimOrder, IterableOrderedOrderSet.QUEUE_START);
+        // assert.equal(auctionDetails.clearingPriceOrder, bytes32(0));
+        // assert.equal(auctionDetails.volumeClearingPriceOrder, 0);
+        // assert.equal(auctionDetails.minFundingThresholdNotReached, false);
+      });
+
+      it("reverts when calling before expiry", async function () {
+        const firstOptionAddress = firstOption.address;
+
+        await vault.connect(ownerSigner).commitAndClose();
+
+        await time.increaseTo((await vault.nextOptionReadyAt()).toNumber() + 1);
+
+        const firstTx = await vault.connect(keeperSigner).rollToNextOption();
+
+        await expect(firstTx)
+          .to.emit(vault, "OpenShort")
+          .withArgs(
+            firstOptionAddress,
+            await collateralContract.balanceOf(MARGIN_POOL),
+            keeper
+          );
+
+        // 100% of the vault's balance is allocated to short
+        assert.bnEqual(
+          await assetContract.balanceOf(vault.address),
+          BigNumber.from(0)
+        );
+
+        await expect(
+          vault.connect(ownerSigner).commitAndClose()
+        ).to.be.revertedWith("C31");
```

Comment:
What is C31? Is it an internal Opyn error?

Reply:
look at line 1223 with controller codes https://etherscan.io/address/0xcc2fd280a669a4453b30b5e72871ad3e2a2663d0/advanced#code

its shorthand for the errors we had before but thats what it reverted with